sopel/irc.py

```diff
@@ -149,6 +149,17 @@ def write(self, args, text=None):
             self.send(temp.encode('utf-8'))
         finally:
             self.writing_lock.release()
+        # Simulate echo-message
+        if ('echo-message' not in self.enabled_capabilities and
+                args[0].upper() in ['PRIVMSG', 'NOTICE']):
+            # Since we have no way of knowing the hostmask the IRC server uses
```

Comment:
That might need some tweak after #1537 is released, I think.

Reply:
s/might/will/, I know. Depends which goes first (should be this PR, as it's much older).